File: package/main/src/Math/random.ts
/**
 * Generates a random integer between min and max (inclusive)
 * @param  {number} max Maximum value (inclusive)
 * @param  {number} min Minimum value (inclusive, default: 0)
 * @returns {number} Random integer between min and max
 * @example random(10); // returns number between 0 and 10
 * @example random(10, 5); // returns number between 5 and 10
 */
export const random = (max: number, min = 0): number =>
  Math.floor(Math.random() * (max - min + 1)) + min;

File: package/umt_rust/src/math/random.rs
use rand::RngExt;

/// Generates a random integer between min and max (inclusive).
///
/// # Arguments
///
/// * `max` - Maximum value (inclusive).
/// * `min` - Minimum value (inclusive, default: 0).
///
/// # Returns
///
/// Random integer between min and max.
///
/// # Examples
///
/// ```
/// use umt_rust::math::umt_random;
///
/// let r = umt_random(10, 0);
/// assert!(r >= 0 && r <= 10);
///
/// let r = umt_random(10, 5);
/// assert!(r >= 5 && r <= 10);
/// ```
pub fn umt_random(max: i64, min: i64) -> i64 {
    let mut rng = rand::rng();
    rng.random_range(min..=max)
}

/// Generates a random integer between 0 and max (inclusive).
///
/// # Arguments
///
/// * `max` - Maximum value (inclusive).
///
/// # Returns
///
/// Random integer between 0 and max.
#[inline]
pub fn umt_random_max(max: i64) -> i64 {
    umt_random(max, 0)
}
